lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -552,62 +562,167 @@
             `|${ALPHA}{4,8}`          +                                         //              ~/ 4*8ALPHA            ; (paraphrased: combined previous two lines)
             ')\\b';
 
-        // below: ${language}, ${script}, and ${region} are wrapped in parens because matching groups are useful for replacement
+        // Use matching groups only when needed
         const LANG_TAG_BASE = `\\b(${language})\\b`         +                   // langtag       = language
                               `\\b(?:-(${script}))?\\b`     +                   //                 ["-" script]
                               `\\b(?:-(${region}))?\\b`     ;                   //                 ["-" region]
-        const LANG_TAG_EXT  = `\\b(?:-${variant})*\\b`      +                   //                 *("-" variant)
+        const LANG_TAG_EXT  = `\\b((?:-${variant})*)\\b`    +                   //                 *("-" variant)
                               `\\b((?:-${extension})*)\\b`  +                   //                 *("-" extension)
-                              `\\b(?:-${privateuse})?\\b`   ;                   //                 ["-" privateuse]
+                              `\\b(?:-(${privateuse}))?\\b` ;                   //                 ["-" privateuse]
```

Comment:
Why does this case have the parens directly around `${foo}` rather than `(?:-${foo})` like the previous 2 cases?

Reply:
because privateuse is 0 or 1 while the previous two were 0 to many -- I can probably change this for consistency's sake, but its only used in parseLangtag, which... now that I think about it, never actually needs privateuse?